xbmc/addons/AddonManager.cpp

```diff
@@ -211,7 +211,11 @@ bool CAddonMgr::ReloadSettings(const std::string &id)
 
 std::vector<std::shared_ptr<IAddon>> CAddonMgr::GetAvailableUpdates() const
 {
-  return GetAvailableUpdatesOrOutdatedAddons(false);
+  std::vector<std::shared_ptr<IAddon>> availableUpdates =
+      GetAvailableUpdatesOrOutdatedAddons(false);
+  m_lastAvailableUpdatesCount = availableUpdates.size();
```

Comment:
CAddonMgr::GetAvailableUpdates() writing the new member and GetLastAvailableUpdatesCount() reading it are accessed from different threads. Thus access to the new member must be mutex protected i both the writing and reading method.

Reply:
ok, regarding the mutex i've searched the webs and come up now with this https://github.com/howie-f/xbmc/commit/1ff3de915f9ed334a19377a29c4784cc6503b835 is this what we need in our case? (for some reason i don't know that guy didn't lock the read in the getter)

Reply:
It's okay as you did. It's a widely spread mistake not to protect the getters. But it's necessary, because the writing thread might be somewhere in the middle of changing data while the unprotected getter reads that half-changed data.

Reply:
BTW: Thre is no reason for stroing a copy of the vector here! Just do

std::string ...::Get....() const
{
  std::string count = std::to_string(GetAvailableUpdatesOrOutdatedAddons(false).size());
  ...mutex lock
  m_lastAva... = count;
  return m_last...
}

Note that m_aval is a string now!

Reply:
yeah, assuming that is `GetLastAvailableUpdateCount()` this would bring our 1700% cpu usage back ;)